main.go

```diff
@@ -374,55 +534,75 @@ func validateBaseURLScheme(baseURL string) (string, error) {
-	flag.StringVar(&token, "Token", "jGWPwqQUuf37b", "Gitlab api key/token")
-	flag.StringVar(&timeInterval, "TimeInterval", "daily", "Set milestone to daily, weekly or monthly")
-	flag.StringVar(&baseURL, "BaseURL", "dev.example.com", "Gitlab api base url")
-	flag.StringVar(&namespace, "Namespace", "someNamespace", "Namespace to use in Gitlab")
-	flag.StringVar(&project, "ProjectName", "someProject", "Project to use in Gitlab")
-	flag.IntVar(&advance, "Advance", 30, "Define timeframe to generate milestones in advance")
+	flag.StringVar(&token, "token", "jGWPwqQUuf37b", "Gitlab api key/token")
+	flag.StringVar(&timeInterval, "time-interval", "daily", "Set milestone to daily, weekly or monthly")
+	flag.StringVar(&baseURL, "base-url", "dev.example.com", "Gitlab api base url")
+	flag.StringVar(&namespace, "namespace", "someNamespace", "Namespace to use in Gitlab")
+	flag.StringVar(&project, "project", "someProject", "Project to use in Gitlab")
+	flag.IntVar(&advance, "advance", 30, "Define timeframe to generate milestones in advance")
 	flag.Parse() //Command Line Parsing Ends
 
 	// Initializing logger
 	LoggerSetup(os.Stdout)
 
-	milestoneData := createMilestoneData(advance, strings.ToLower(timeInterval))
-
 	// Validate baseURL scheme
 	URL, err := validateBaseURLScheme(baseURL)
 	if err != nil {
 		logger.Println(err)
 	}
 
-	// Calling getProjectID
-	baseURL = URL + "/api/v4"
-	projectID, err := getProjectID(baseURL, token, project, namespace)
+	// Check which API to use
+	api, err := checkAPI(URL, token, namespace, project)
 	if err != nil {
 		logger.Fatal(err)
-		// TODO: check for authentication error (currently it only says project not found)
-	}
-	err = createAndDisplayNewMilestones(baseURL, token, projectID, milestoneData)
-	if err != nil {
-		logger.Println(err)
 	}
-	editMilestones, err := getClosedMilestones(baseURL, token, projectID, milestoneData)
-	if err != nil {
-		logger.Println(err)
-	}
-	err = reactivateClosedMilestones(editMilestones, baseURL, token, projectID)
-	if err != nil {
-		logger.Println(err)
+	milestoneData := createMilestoneData(advance, strings.ToLower(timeInterval), api)
+
+	// Calling getProjectID
+	var newBaseURL, projectID string
+	switch {
+	case api == "gitlab":
+		newBaseURL = URL + "/api/v4"
+		projectID, err = getProjectID(newBaseURL, token, project, namespace, api)
+		if err != nil {
+			logger.Fatal(err)
+		}
+		err = createAndDisplayNewMilestones(newBaseURL, token, projectID, milestoneData, api)
+		if err != nil {
+			logger.Println(err)
+		}
+		editMilestones, err := getClosedMilestones(newBaseURL, token, projectID, milestoneData, api)
+		if err != nil {
+			logger.Println(err)
+		}
+		err = reactivateClosedMilestones(editMilestones, newBaseURL, token, projectID, api)
+		if err != nil {
+			logger.Println(err)
+		}
+	case api == "github":
+		newBaseURL = URL + "/repos/" + namespace + "/" + project
+		err = createAndDisplayNewMilestones(newBaseURL, token, "", milestoneData, api)
```

Comment:
Might be better to pass the project here instead, and append it inside the function as we do for project ID for GitLab?

Reply:
Same as my above comment - GitHub doesn't use a project ID to get milestones and instead is tied to the repo/issues

Reply:
But could the project name be passed in the `projectID` field of the variable ? That way it looks more homogeneous with how it's done for gitlab (That would be, passing in `project` instead of `""` here and append it inside the function)

It doesn't change anything from a logic perspective but it keeps things similar for gitlab and github ; if in the future we change something about how we handle the building of the URL we'll probably miss the fact that depending on the API the project/projectID is added inside or outside some functions.

Reply:
Ok, I made that change and renamed "projectID" to "project" since we're actually passing in the project name for GitHub.